### PR TITLE
Ensure versions of jb/wfly match java compatibility

### PR DESCRIPTION
**Pull request by robstryker**
Signed-off-by: Rob Stryker <rob@oxbeef.net>

### CODE REVIEW

**Comment by adietish** [comment]
testPR

File: framework/bundles/org.jboss.tools.rsp.server.spi/src/main/java/org/jboss/tools/rsp/server/spi/util/VersionComparisonUtility.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+		if( vmiMajor > testMajor ) return false;
+		if( vmiMajor < testMajor ) return true;
+		
+		if( splitTest.length <= 1 || splitTest[1].isEmpty())
+			return true;
+
+		// Majors are equal. 
+		int vmiMinor = Integer.parseInt(splitVmi[1]);
+		int testMinor = Integer.parseInt(splitTest[1]);
+		if( vmiMinor > testMinor ) return false;
```
Comment (adietish) [inline framework/bundles/org.jboss.tools.rsp.server.spi/src/main/java/org/jboss/tools/rsp/server/spi/util/VersionComparisonUtility.java]: sonar: l49-l50 can be replaced by a single return statement, no need for if/else
```return vmiMinor < testMinor;```

File: framework/bundles/org.jboss.tools.rsp.server.spi/src/main/java/org/jboss/tools/rsp/server/spi/util/VersionComparisonUtility.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		int testMajor = Integer.parseInt(splitTest[0]);
+		if( vmiMajor < testMajor ) return false;
+		if( vmiMajor > testMajor ) return true;
+		
+		// Majors are equal. 
+		if( splitTest.length <= 1 || splitTest[1].isEmpty())
+			return true;
+		int vmiMinor = Integer.parseInt(splitVmi[1]);
+		int testMinor = Integer.parseInt(splitTest[1]);
+		if( vmiMinor < testMinor ) return false;
```
Comment (adietish) [inline framework/bundles/org.jboss.tools.rsp.server.spi/src/main/java/org/jboss/tools/rsp/server/spi/util/VersionComparisonUtility.java]: sonar: l28-l29 can be replaced by a single return statement, no need for if/else
return vmiMinor > testMinor;

Comment (robstryker) [reply framework/bundles/org.jboss.tools.rsp.server.spi/src/main/java/org/jboss/tools/rsp/server/spi/util/VersionComparisonUtility.java]: When I have several conditions, one after another, with multiple possible returns, I really don't like having the last line in the method be an on-the-fly return calculation.

**Pull request merged**